### PR TITLE
feat: implement call page

### PR DESCRIPTION
**Pull request by smeijer**
Added color and content to the `/call` page. The design is inspired by `/chats` (podcast), but instead of opening every call in a details page, they expand and show the player inline.

**TODOS**

- Alex is rendered, but in low quality. We need higher resolution images

**IDEAS**

_copied from Discord:_

- Add search (like on blog page)
- Add keyword filter (like on blog page)
- Display summary on each row, and the description in expanded view?
- Display the keywords on each row as well?

### CODE REVIEW

File: app/routes/call/$slug.tsx [hunk 1]
```diff
@@ -24,12 +25,17 @@ export default function Screen() {
...
+
   return (
-    <div>
-      <pre>{JSON.stringify(episode, null, 2)}</pre>
-    </div>
+    <div
+      dangerouslySetInnerHTML={{
+        __html: theme === 'dark' ? episode.embedHtmlDark : episode.embedHtml,
+      }}
+    />
```
Comment (smeijer) [inline app/routes/call/$slug.tsx:39]: Note that this doesn't work with fake data, as the iframe then doesn't load. Replace with the below for testing purpose:

`<iframe src='https://share.transistor.fm/e/1493e91f/dark' width='100%' height='180' frameborder='0' scrolling='no' seamless='true' style='width:100%; height:180px;'></iframe>`.

File: app/routes/call.tsx [hunk 2]
```diff
@@ -20,30 +34,183 @@ export const loader: LoaderFunction = async ({request}) => {
...
+    sortOrder === 'desc' ? data.episodes : [...data.episodes].reverse()
+
+  // An effect to scroll to the episode's position when opening a direct link
+  React.useEffect(() => {
+    if (activeSlug) {
+      document.querySelector(`[data-episode="${activeSlug}"]`)?.scrollIntoView()
+    }
+    // we really only want to run this on mount. Otherwise it will hijack the
+    // users scroll while they browse the episodes
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```
Comment (kentcdodds) [inline app/routes/call.tsx:68]: In this case we can use a ref so we don't have to disable the rule

File: app/routes/call/$slug.tsx (hunk 1 above)
Comment (kentcdodds) [reply app/routes/call/$slug.tsx:39]: We could update the mocks in `mocks/transistor.ts`

File: app/routes/call.tsx [hunk 3]
```diff
@@ -20,30 +34,183 @@ export const loader: LoaderFunction = async ({request}) => {
...
+                            opacity: 1,
+                          },
+                        }}
+                        initial="collapsed"
+                        animate="expanded"
+                        exit="collapsed"
+                        transition={{duration: 0.15}}
+                        className="relative col-span-full"
+                      >
+                        <CallKentEpisodesProvider value={data.episodes}>
```
Comment (kentcdodds) [inline app/routes/call.tsx:196]: Maybe we should move this out to wrap the whole screen so we don't have to render it for every episode

File: app/routes/call.tsx [hunk 4]
```diff
@@ -20,30 +34,183 @@ export const loader: LoaderFunction = async ({request}) => {
...
+              <>
+                Showing newest first
+                <ChevronDownIcon className="ml-2 text-gray-400" />
+              </>
+            )}
+          </button>
+        </div>
+
+        <div className="col-span-full">
+          {sortedEpisodes.map((episode, idx, {length}) => {
```
Comment (kentcdodds) [inline app/routes/call.tsx:128]: Should we try and use `@reach/accordion` for this so we don't have to worry about making sure it's accessible?

File: app/routes/call.tsx [hunk 5]
```diff
@@ -20,30 +34,183 @@ export const loader: LoaderFunction = async ({request}) => {
...
+                      </div>
+                      <img
+                        className="w-full rounded-lg object-cover"
+                        src={episode.imageUrl}
+                        alt={episode.title}
+                      />
+                    </div>
+                    <div className="text-primary relative flex flex-col col-span-3 md:col-span-7 lg:flex-row lg:col-span-11 lg:items-center lg:justify-between">
+                      <div className="mb-3 text-xl font-medium lg:mb-0">
+                        {/* TODO: make it three digits? How many calls do we expect? */}
```
Comment (kentcdodds) [inline app/routes/call.tsx:158]: Yeah, let's be optimistic and do three digits 😅

File: app/routes/call.tsx [hunk 6]
```diff
@@ -20,30 +34,183 @@ export const loader: LoaderFunction = async ({request}) => {
...
+                        <span className="inline-block w-10 lg:text-lg">
+                          {`${number.toString().padStart(2, '0')}.`}
+                        </span>
+
+                        {episode.title}
```
Comment (kentcdodds) [inline app/routes/call.tsx:163]: Could/should we also display the episode summary and keywords in a small font here? I'm not sure about the summary, but keywords might be useful. The problem is, I'm not sure what the summary will be. Right now we don't connect that information anyway...

**Review by kentcdodds — COMMENTED** [review]
Awesome! Man I'm really amazed. Just a few things

File: app/routes/call.tsx (hunk 3 above)
Comment (smeijer) [reply app/routes/call.tsx:196]: Makes sense. Thanks!

File: app/routes/call.tsx (hunk 5 above)
Comment (smeijer) [reply app/routes/call.tsx:158]: I've made them dynamic. :sunglasses: It now shows as many leading zeros as required, based on the number of episodes available. You're good until episode 9999. Probably that's the right time to add some windowing or pagination as well :sweat_smile:

Comment (kentcdodds) [reply app/routes/call.tsx:158]: I like your idea of splitting things by year 👍 We'll see how things look if/when we get to a ton of episodes :)

File: app/routes/call.tsx (hunk 6 above)
Comment (smeijer) [reply app/routes/call.tsx:163]: With summary you mean `description` right? There is no limit on how long that can get. That's why I showed it inside the accordion.

I'll add the keywords. But there also, there is no limit on how many. Should we stick to the first 3? Separated by space?

File: app/routes/call.tsx (hunk 4 above)
Comment (smeijer) [reply app/routes/call.tsx:128]: I'm not sure. It's just a `link` that navigates to a new page. Does it make sense to use accordion for that? We use it in `/discord`, but the accordions there don't navigate. Does that make a difference?

Happy to add it. Just wanna make sure that there is value in it.

File: app/routes/call.tsx (hunk 6 above)
Comment (smeijer) [reply app/routes/call.tsx:163]: Maybe even just add both in the accordion. Looks quite nice!

image

Comment (kentcdodds) [reply app/routes/call.tsx:163]: Why don't we put the keywords (unlabeled) in muted text below the title? I just want to make it easy to scan the episodes. Maybe that's not necessary though 🤔

File: app/routes/call.tsx (hunk 4 above)
Comment (kentcdodds) [reply app/routes/call.tsx]: As long as it's keyboard accessible then I think we're good.

File: app/routes/call.tsx (hunk 6 above)
Comment (smeijer) [reply app/routes/call.tsx]: I've created #77. Let's take another look at that asap.

**Pull request merged**